ngraph/core/shape_inference/include/scatter_nd_base_shape_inference.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/util/scatter_nd_base.hpp>
+
+#include "utils.hpp"
+
+template <class T>
+void shape_infer(const ov::op::util::ScatterNDBase* op,
+                 const std::vector<T>& input_shapes,
+                 std::vector<T>& output_shapes,
+                 const std::map<size_t, std::shared_ptr<ngraph::runtime::HostTensor>>& constant_data = {}) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 3 && output_shapes.size() == 1);
+    const auto& inputs_shape = input_shapes[0];
+    const auto& indices_shape = input_shapes[1];
+    const auto& updates_shape = input_shapes[2];
+
+    const auto& inputs_rank = inputs_shape.rank();
+    const auto& indices_rank = indices_shape.rank();
+    const auto& updates_rank = updates_shape.rank();
+
+    NODE_VALIDATION_CHECK(op,
+                          indices_rank.is_dynamic() || indices_rank.get_length() >= 1,
```

Comment:
basically you are requiring `indices_rank != 0`
Please see -- [Rank is just a Dimension](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/openvino/core/rank.hpp) and [Dimension has != operator overloaded](https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/openvino/core/dimension.hpp#L40)